backend/assignments/1/src/main/java/org/example/Assignment/ExecuteTransaction.java

```diff
@@ -0,0 +1,197 @@
+package org.example.Assignment;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+
+
+public class ExecuteTransaction implements Runnable {
+
+    private JsonNode transaction;
+    private CountDownLatch latch;
+    private Map<String, Coin> coinMap;
+    private Map<String, Trader> traderMap;
+    private String transactionType;
+
+    public ExecuteTransaction() {
+
+    }
+
+    public ExecuteTransaction(JsonNode transaction, CountDownLatch latch, Map<String, Coin> coinMap, Map<String, Trader> traderMap) {
+        this.transaction = transaction;
+        this.coinMap = coinMap;
+        this.traderMap = traderMap;
+        this.transactionType = transaction.get("type").asText();
+        this.latch = latch;
+    }
+
+    private boolean buyingFunction() {
+        String coinSymbol = transaction.get("data").get("coin").asText();
+        Coin coinone = coinMap.get(coinSymbol);
+
+        if (coinone == null) {
+            // Handle the case when the coin is not found in the map
+            return false;
+        }
+
+
+        long availableVolume = coinMap.get(transaction.get("data").get("coin").asText()).getCirculatingSupply();
+
+        long quantity = transaction.get("data").get("quantity").asLong();
+
+        if (quantity > availableVolume) return false;
+
+        synchronized (coinMap.get(transaction.get("data").get("coin").asText())) {
+            long difference = availableVolume - quantity;
+            coinMap.get(transaction.get("data").get("coin").asText()).setCirculatingSupply(difference);
+        }
+        synchronized (traderMap.get(transaction.get("data").get("wallet_address").asText())) {
+            Trader trader = traderMap.get(transaction.get("data").get("wallet_address").asText());
+            Coin coin = coinMap.get(transaction.get("data").get("coin").asText());
+
+            if (trader.getPortfolio().coinQuantities.containsKey(coin.getSymbol())) {
+                Long available = trader.getPortfolio().coinQuantities.get(coin.getSymbol());
+                Long updated = available + quantity;
+                trader.getPortfolio().coinQuantities.put(coin.getSymbol(), updated);
+            } else {
+                trader.getPortfolio().coinQuantities.put(coin.getSymbol(), quantity);
+            }
+
+            if (trader.getPortfolio() != null) {
```

Comment:
`if(!something)` can suffice.
Also, using `Objects.isNull()` for null checks is cleaner